core/indigo-core/molecule/src/base_molecule.cpp

```diff
@@ -4325,3 +4313,81 @@ void BaseMolecule::getBoundingBox(Rect2f& bbox) const
     }
     bbox = Rect2f(a, b);
 }
+
+void MetaData::addMetaObject(MetaObject* pobj)
+{
+    int index = _meta_data.size();
+    _meta_data.expand(index + 1);
+    _meta_data.set(index, pobj);
+
+    switch (pobj->_class_id)
+    {
+    case KETSimpleObject::CID:
+        _simple_object_indexes.push() = index;
+        break;
+    case KETReactionPlus::CID:
+        _plus_indexes.push() = index;
+        break;
+    case KETReactionArrow::CID:
+        _arrow_indexes.push() = index;
+        break;
+    default:
+        break;
+    }
+}
+
+void MetaData::cloneMetaData(const MetaData& other)
+{
+    resetMetaData();
+    const auto& meta = other.metaData();
+    for (int i = 0; i < meta.size(); i++)
+        addMetaObject(meta[i]->clone());
+}
```

Comment:
Why do we keep using this clone methods instead of copy constructors?

Reply:
Unfortunately we still can't use std classes as members in Indigo.

Reply:
I understand the reason why we can't use `std`, but why we can't use copy constructors instead of `clone` methods?